libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1152,6 +1174,35 @@ std::variant<StandardCompiler::InputsAndSettings, Json> StandardCompiler::parseI
 		ret.modelCheckerSettings.timeout = modelCheckerSettings["timeout"].get<Json::number_unsigned_t>();
 	}
 
+	if ((ret.debugInfoSelection.has_value() && ret.debugInfoSelection->ethdebug) || isEthdebugRequested(ret.outputSelection))
+	{
+		if (ret.language != "Solidity" && ret.language != "Yul")
+			return formatFatalError(Error::Type::FatalError, "'settings.debug.debugInfo' 'ethdebug' is only supported for languages 'Solidity' and 'Yul'.");
+	}
+
+	if (isEthdebugRequested(ret.outputSelection))
+	{
+		if (ret.language == "Solidity" && !ret.viaIR)
+			return formatFatalError(Error::Type::FatalError, "'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' can only be selected as output, if 'viaIR' was set.");
+
+		if (!ret.debugInfoSelection.has_value())
+		{
+			ret.debugInfoSelection = DebugInfoSelection::Default();
+			ret.debugInfoSelection->enable("ethdebug");
+		}
+		else
+		{
+			if (!ret.debugInfoSelection->ethdebug && ret.language == "Solidity")
+				return formatFatalError(Error::Type::FatalError, "'ethdebug' needs to be enabled in 'settings.debug.debugInfo', if 'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' was selected as output.");
+		}
+	}
+
+	if (
+		ret.debugInfoSelection.has_value() && ret.debugInfoSelection->ethdebug && ret.language == "Solidity" &&
+		!pipelineConfig(ret.outputSelection)[""][""].irCodegen && !isEthdebugRequested(ret.outputSelection)
+	)
+		return formatFatalError(Error::Type::FatalError, "'settings.debug.debugInfo' can only include 'ethdebug', if output 'ir', 'irOptimized', 'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' was selected.");
```

Comment:
```suggestion
		return formatFatalError(Error::Type::FatalError, "'settings.debug.debugInfo' can only include 'ethdebug', if output 'ir', 'irOptimized', 'evm.bytecode.ethdebug', or 'evm.deployedBytecode.ethdebug' was selected.");
```

Aside from the comma - shouldn't it be that you have to select one of `ir` and `irOptimized` as well as one of `evm.bytecode.ethdebug` and `evm.deployedBytecode.ethdebug`? This reads to me as if one of the four is sufficient.

Reply:
This is about the debug info selection. If `ethdebug` was selected as a debug info artefact, it will only work, if either `ir` or `irOptimzed` (`--ir` or `--ir-optimized` in CLI world) was selected as output. However, it is also valid with `evm.bytecode.ethdebug` or `evm.deployedBytecode.ethdebug` (`--ethdebug` or `--ethdebug-runtime` in CLI) was selected. Note that those are only valid, if via-ir was defined. E.g.  `evm.bytecode.ethdebug` or `evm.deployedBytecode.ethdebug` (`--ethdebug`and `--ethdebug-runtime`) can only be used, if via-ir was enabled.

Reply:
ok I see, thanks for the clarifications!